embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpanImpl.kt

```diff
@@ -23,23 +22,17 @@ import java.util.concurrent.atomic.AtomicReference
 internal class EmbraceSpanImpl(
     private val spanName: String,
     private val openTelemetryClock: Clock,
-    private val spanBuilder: SpanBuilder,
+    private val spanBuilder: EmbraceSpanBuilder,
     override val parent: EmbraceSpan? = null,
-    private val spanRepository: SpanRepository? = null,
-    sessionSpan: Boolean = false
+    private val spanRepository: SpanRepository? = null
 ) : PersistableEmbraceSpan {
 
-    init {
```

Comment:
Reimplemented the logic to mark key spans at the EmbraceSpanBuilder wrapper level and change the criteria to explicitly do that for perf emb.type with null parents rather than doing that for all spans with null parents that are not session spans. Getting rid of this lil hack felt good!